src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -0,0 +1,54 @@
+package checkers.inference.solver.frontend;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+/**
+ * Special Lattice class for two qualifier type system.
+ * 
+ * @author jianchu
+ *
+ */
+public class TwoQualifiersLattice extends Lattice {
+
+    protected Set<AnnotationMirror> allTypes;
```

Comment:
Um... I'm not very sure... Maybe `Set<AM>` is enough?

For `QualifierHierarchy`, it seems declares the method `getTypeQuals()` return `Set<? extends AM>` in order to let sub-classes have flexibility on giving stronger restrictions.

Will we have such needs for Lattice? If not, maybe `Set<AM>` is enough.